vowpalwabbit/core/src/parse_args.cc

```diff
@@ -406,6 +406,26 @@ input_options parse_source(VW::workspace& all, options_i& options)
                      "hashed as A^B^C."))
       .add(make_option("flatbuffer", parsed_options.flatbuffer)
                .help("Data file will be interpreted as a flatbuffer file")
+               .experimental())
+      .add(make_option("csv", parsed_options.csv).help("Data file will be interpreted as a csv file").experimental())
+      .add(make_option("separator", all.separator)
+               .short_name("s")
+               .help("CSV Parser: Explicitly specify field separator (Perl-regex) "
```

Comment:
Oh, actually I wasn't intended to support the regex, but the arbitrary single separator. The comment is just directly copied from https://github.com/HollowMan6/vowpal_wabbit/blob/master/utl/csv2vw#L76 and I didn't pay much attention to it, sorry about that.

Well, for normal CSV files, the separators are usually commas, and we also use commas here as the default separator, users don't need to specify that option for normal CSV files with commas separator. In addition, I guess it's better to support custom separators, there's much more separators (such as spaces etc.) instead of only tabs and commas are available, with this option, users can have more flexibility. Also, automatically judging if the separator is tabs or commas may lead to problems and make the parser become completely unusable when our algorithms has bugs to judging it and separate it wrongly, as you can't always count on such algorithms.

Reply:
Yep, that's fair. Supporting other separators makes sense. I do think we should constrain to single characters to avoid complexity.

I do not think we should ever try auto-detecting the separator. It is a minefield I agree.